### PR TITLE
Fix build machine and compiler configuration files on summit with IBM compiler

### PR DESCRIPTION
**Pull request by xyuan**
Adding a new module dependency (xalt/1.1.4) for the XL compiler. Without this change netcdf/pnetcdf modules fail to load with the XL compiler

[BFB]

### CODE REVIEW

**Comment by xyuan** [comment]
1): add "load xalt/1.1.4" before loading xlf model to avoid the module load errors. 
2): add option "-qrealsize=8" for double precision build.

**Comment by brhillman** [comment]
@xyuan does this just fix the non-MMF build? You will need to make more fixes, correct?

**Comment by xyuan** [comment]
@brhillman this modifications only make the code to be able to build, the source codes also need to fix some issues, such as type mismatch, and some fortran grammar error. I leave these to the developer.

**Comment by whannah1** [comment]
@xyuan can you create a github issue to describe these type mismatch and grammar issues?

**Comment by whannah1** [comment]
@xyuan I don't think the "enhancement" and "ECPP" labels apply to this PR. It seems to only be a fix to the machine config and not an additional feature and has nothing to do with aerosols. So you should remove those labels.

**Comment by xyuan** [comment]
> @xyuan can you create a github issue to describe these type mismatch and grammar issues?

yes, will do it tomorrow.

File: cime/config/e3sm/machines/config_compilers.xml [hunk 1]
```diff
@@ -1791,6 +1791,9 @@ ntel/x86_64/2013/composer_xe_2013/composer_xe_2013_sp1.3.174/mkl/include </base>
   <NETCDF_C_PATH>$ENV{NETCDF_C_PATH}</NETCDF_C_PATH>
   <NETCDF_FORTRAN_PATH>$ENV{NETCDF_FORTRAN_PATH}</NETCDF_FORTRAN_PATH>
   <SUPPORTS_CXX>TRUE</SUPPORTS_CXX>
+  <FC_AUTO_R8>
+    <base> -qrealsize=8 </base>
+  </FC_AUTO_R8>
```
Comment (jayeshkrishna) [inline cime/config/e3sm/machines/config_compilers.xml]: Is this change on E3SM master? Why did you need this change?

**Comment by jayeshkrishna** [comment]
@xyuan : Can you modify the commit message (the commit that changed the code - requires rebase of your branch) or the PR description (which will be used as the commit message for the merge commit) to include more information on this change ?

(e.g. Adding a new module dependency (xalt/1.1.4) for the XL compiler. Without this change netcdf/pnetcdf modules fail to load with the XL compiler)

File: cime/config/e3sm/machines/config_compilers.xml (hunk 1 above)
Comment (xyuan) [reply cime/config/e3sm/machines/config_compilers.xml]: > Is this change on E3SM master? Why did you need this change?

add -qrealsize=8 promotes the default byte size for REAL variables declared without an explicit byte size which will improve the robustness of the code.

Comment (optimus-prime) [reply cime/config/e3sm/machines/config_compilers.xml]: We recommend against auto-promotion of variables in E3SM.

Comment (sarats) [reply cime/config/e3sm/machines/config_compilers.xml]: We recommend against auto-promotion of variables in E3SM.

Comment (xyuan) [reply cime/config/e3sm/machines/config_compilers.xml]: > We recommend against auto-promotion of variables in E3SM.

sorry, if you check both the e3sm and acme config_compilers.xml, there are r8 auto-promotion option activated for many compilers.

Comment (rljacob) [reply cime/config/e3sm/machines/config_compilers.xml]: On E3SM master, realsize only appears in one place: the generic ibm config compilers entry.

Our goal is to have all fortran variables and constants declare their precision explicitly so -qrealisize should have no effect/not be needed.

Comment (xyuan) [reply cime/config/e3sm/machines/config_compilers.xml]: > On E3SM master, realsize only appears in one place: the generic ibm config compilers entry.
> 
> Our goal is to have all fortran variables and constants declare their precision explicitly so -qrealisize should have no effect/not be needed.

Nice comments, this is much better than auto-protomotion options,  I am going to close this PR, and create a new one without this option. Thanks very much.

**Comment by xyuan** [comment]
close this PR to create a new PR without this auto-promotion option.
